tcp-source/src/test/java/org/springframework/cloud/stream/module/tcp/source/TcpSourceTests.java

```diff
@@ -0,0 +1,198 @@
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.springframework.cloud.stream.module.tcp.source;
+
+import static org.hamcrest.CoreMatchers.is;
+import static org.junit.Assert.assertEquals;
+import static org.junit.Assert.assertFalse;
+import static org.junit.Assert.assertThat;
+import static org.junit.Assert.assertTrue;
+import static org.springframework.cloud.stream.test.matcher.MessageQueueMatcher.receivesPayloadThat;
+
+import java.net.Socket;
+
+import javax.net.SocketFactory;
+
+import org.hamcrest.Matchers;
+import org.junit.Test;
+import org.junit.runner.RunWith;
+
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.boot.test.IntegrationTest;
+import org.springframework.boot.test.SpringApplicationConfiguration;
+import org.springframework.cloud.stream.messaging.Source;
+import org.springframework.cloud.stream.test.binder.MessageCollector;
+import org.springframework.integration.ip.tcp.connection.AbstractServerConnectionFactory;
+import org.springframework.integration.ip.tcp.connection.TcpNetServerConnectionFactory;
+import org.springframework.integration.ip.tcp.connection.TcpNioServerConnectionFactory;
+import org.springframework.integration.test.util.TestUtils;
+import org.springframework.test.annotation.DirtiesContext;
+import org.springframework.test.context.junit4.SpringJUnit4ClassRunner;
+
+/**
+ * Tests for TcpSource.
+ *
+ * @author Gary Russell
+ */
+@RunWith(SpringJUnit4ClassRunner.class)
+@SpringApplicationConfiguration(classes = TcpSourceApplication.class)
+@DirtiesContext
+public abstract class TcpSourceTests {
+
+	@Autowired
+	protected Source channels;
+
+	@Autowired
+	protected MessageCollector messageCollector;
+
+	@Autowired
+	protected AbstractServerConnectionFactory connectionFactory;
+
+	@Autowired
+	protected TcpSourceProperties properties;
+
+	@IntegrationTest({ "port = 0", "nio = true", "reverseLookup = true",
+					"useDirectBuffers = true", "socketTimeout = 123" })
+	public static class PropertiesPopulatedTests extends TcpSourceTests {
+
+		@Test
+		public void test() throws Exception {
+			assertThat(this.connectionFactory, Matchers.instanceOf(TcpNioServerConnectionFactory.class));
+			assertTrue(TestUtils.getPropertyValue(this.connectionFactory, "lookupHost", Boolean.class));
+			assertTrue(TestUtils.getPropertyValue(this.connectionFactory, "usingDirectBuffers", Boolean.class));
+			assertEquals(123, TestUtils.getPropertyValue(this.connectionFactory, "soTimeout"));
+		}
+
+	}
+
+	@IntegrationTest({ "port = 0" })
```

Comment:
You can move `port` option to the `TcpSourceTests` level:
```
@WebIntegrationTest("port = 0")
```
and remove it from each these particular `@IntegrationTest`.
See `CassandraSinkIntegrationTests`.